SSC updates based on feedback (#822)
returntocorp/semgrep-docs: docs/semgrep-sc/sc-getting-started.md

}
</ul>

# Getting started with Semgrep Supply Chain

> Semgrep Supply Chain detects recently discovered [security vulnerabilities](https://nvd.nist.gov/vuln/full-listing) through [reachability](/docs/semgrep-sc/sc-glossary#reachability) analysis. 

Reviewer: detecting via reachability analysis might be misleading, maybe can be something like 

"SSC detects recently discovered security vulnerability in your code, prioritizing urgent ones via reachability analysis"

Author: Added! Thank yeew. Merging.

> Semgrep Supply Chain detects recently discovered [security vulnerabilities](https://nvd.nist.gov/vuln/full-listing) in your codebase's open source dependencies, prioritizing findings through [reachability](/docs/semgrep-sc/sc-glossary#reachability) analysis. 

Reviewer 2: ```suggestion
Semgrep Supply Chain detects recently discovered [security vulnerabilities](https://nvd.nist.gov/vuln/full-listing) in your codebase's open source dependencies, prioritizing findings through [reachability](/semgrep-sc/sc-glossary#reachability) analysis. 
```

This document walks you through the process of setting up open source dependency scanning with Semgrep Supply Chain.

:::note

> Semgrep Supply Chain **supports monorepositories (monorepos)** by treating each subdirectory as its own repository. Findings are grouped under these repositories based on the [lockfile](/docs/semgrep-sc/sc-glossary/#lockfile) or manifest file present in the subdirectory.

Reviewer 2: Just a small note - for absolute internal links (which work the best when we are changing the structure of the docs, so a good choice), do not include the `/docs/` in the beginning. The root for links in `/` is considered as `/docs/` already. Otherwise, this causes some issues which you can see if you run `yarn build && yarn serve`. 🙂

```suggestion
Semgrep Supply Chain **supports monorepositories (monorepos)** by treating each subdirectory as its own repository. Findings are grouped under these repositories based on the [lockfile](/semgrep-sc/sc-glossary/#lockfile) or manifest file present in the subdirectory.
```

This is the preferred method to enable and run Semgrep Supply Chain.
:::

> Semgrep Supply Chain scans can be set up from the [Semgrep App](/) interface.

Reviewer 2: @s-santillan just checking if this is not an omission, feel free to dismiss this comment. The `/` links to the hope page (resolve if this was the intention here). Maybe you wanted to link this to Semgrep App docs?

Author: Good catch, i'll fix it.

<PlatformSigninGithub />

> See [Permissions in GitHub](/docs/semgrep-app/getting-started-with-semgrep-app/#permissions-for-github) to learn more about how Semgrep features use requested permissions in GitHub.

Reviewer 2: ```suggestion
See [Permissions in GitHub](/semgrep-app/getting-started-with-semgrep-app/#permissions-for-github) to learn more about how Semgrep features use requested permissions in GitHub.
```

Author: Hope we can make vale.sh lint the docs one day so we don't have to do this...

3. Click <i class="fa-solid fa-toggle-large-on"></i> **SSC toggle**. Semgrep Supply Chain rules are included in your next scan, that occurs based on your CI set up, such as schedules and events configuration (push, pull, and merge requests).
4. Optional: Some CI providers enable you to start workflows manually. To start a dependency scan immediately, go to your CI provider's interface and manually begin the Semgrep workflow or job.

> When the scan finishes, Semgrep App displays an overview of findings in the **Supply Chain** tab for further triage and remediation. See [Triaging and remediating dependency findings](/docs/semgrep-sc/triaging-and-remediating-vulnerabilities).

Reviewer 2: ```suggestion
When the scan finishes, Semgrep App displays an overview of findings in the **Supply Chain** tab for further triage and remediation. See [Triaging and remediating dependency findings](/semgrep-sc/triaging-and-remediating-vulnerabilities).
```

Both reachable and unreachable findings of Semgrep Supply Chain do **not** block a pull request or merge request.

:::info
Semgrep versions v0.122.0 and below previously blocked reachable findings.
:::

### Step 4: Set a daily scan schedule

> Semgrep Supply Chain frequently releases new rules. To ensure that your repository is scanned with the latest Semgrep Supply Chain rules, **ensure that Semgrep Supply Chain scans your codebase every day**. 

Reviewer 2: Non-blocking and optional:
```suggestion
Semgrep Supply Chain frequently releases new rules. To ensure that your repository is scanned with the latest Semgrep Supply Chain rules, **configure Semgrep Supply Chain to scan your codebase every day**. 
```

## Triaging and remediating dependency findings

> Semgrep Supply Chain enables developers to perform triage and remediation through the **[Vulnerabilities](https://semgrep.dev/orgs/-/supply-chain/vulnerabilities)** page. On this page you can perform the following:

Reviewer 2: Non-blocking and optional:
```suggestion
Semgrep Supply Chain enables developers to perform triage and remediation through the **[Vulnerabilities](https://semgrep.dev/orgs/-/supply-chain/vulnerabilities)** page. On this page you can perform the following actions:
```
